doc/tutorials/notifications.md

```diff
@@ -0,0 +1,327 @@
+should be used.
+ 
+ In the following example we will register a integer
+variable:
+ 
+```C
+KDB * repo;
+
+//  ... initialization of KDB and I/O-binding
+
+Key * someKey = keyNew ("/sw/someprogram/v1/someKey", KEY_END);
+int someKeyValue;
+
+elektraNotificationRegisterInt (repo, &someKeyValue, someKey);
+```
+ 
+ The variable `someKeyValue` will be
+automatically updated if the Key in the program above is changed by
+another program (e.g. by using the `kdb` CLI command).
+ 
+ The following
+snippet shows how a callback can be used if the value of the changed Key
+needs further processing.
+ 
+```C
+void myCallback (Key * changedKey)
+{
+		// do something
+}
+
+int main ()
+{
+		KDB * repo;
+
+		// ... initialization of KDB and I/O-binding
+
+		Key * someKey = keyNew ("/sw/someprogram/v1/someKey", KEY_END);
+
+		elektraNotificationRegisterCallback(repo, &myCallback, someKey);
+
+		// ...
+}
+```
+ 
+ How to create your own I/O-binding
+ ===
+ 
+ Developers can
+create their own bindings if the I/O management library of their
+choice is not supported by an existing I/O-binding.
+ 
+ To create a
+binding the I/O-interface's functions pointers need to be filled.
+This interface is a `struct` that contains pointers to functions
+for managing file descriptors, timers and idle operations.
+ The
+interface is independent from I/O management libraries and notification
+transport plugins.
+ See the developer API documentation for details on the
+[ElektraIoInterface](https://doc.libelektra.org/api/current/html/structElektraIoInterface.html).
+
+ 
+ 
+```C
+#include <kdbio.h>
+
+// It is helpful to create a data structure for your binding to store additional data
+typedef struct MyBindingOperationData
+{
+	ElektraIoInterface * binding;
+	// Add additional fields as required
+	char* bar;
+} MyBindingData;
+
+
+/**
+ * Add idle task to IO-Binding
+ * @see kdbio.h ElektraIoInterface::addIdle
+ */
+static int myElektraIoBindingAddIdle (ElektraIoInterface * binding, ElektraIoIdleOperation * idleOp)
+{
+	if (idleOp->callback == NULL)
+	{
+		return -1;
+	}
+
+	MyBindingOperationData * myBindingData = malloc (sizeof (*myBindingData));
+	if (bindingData == NULL)
+	{
+		return -1;
+	}
+	memset (bindingData, 0, sizeof (*bindingData));
+
+	idleOp->binding = binding;
+	// you can store additional data for each operation in the bindingData field
+	idleOp->bindingData = myBindingData;
+	myBindingData->binding = binding;
+	myBindingData->bar = "foo";
+
+	// TODO: add idle operation to the I/O management library
+
+	return 0;
+}
+
+/**
+ * Create and initialize a new IO binding.
+ * @param  loop Loop to use for IO operations
+ * @return      Populated IO interface
+ */
+ElektraIoInterface * myElektraIoBindingNew (void /* data from I/O management library */ )
+{
+	ElektraIoInterface * binding = elektraMalloc (sizeof (*binding));
+	if (binding == NULL)
+	{
+		return NULL;
+	}
+	// Set function pointers
+	binding->addFd = myElektraIoBindingAddFd;
```

Comment:
Where do we get this from?

Example seems to be too big for this tutorial. Maybe add something similar to src/plugins/doc?

Reply:
Yes the the full code would be too big. Having some kind of "doc"-binding is a good idea.